src/backend/distributed/commands/function.c

```diff
@@ -1340,18 +1330,80 @@ PostprocessCreateFunctionStmt(Node *node, const char *queryString)
 		return NIL;
 	}
 
-	ObjectAddress address = GetObjectAddressFromParseTree((Node *) stmt, false);
-	EnsureDependenciesExistOnAllNodes(&address);
+	ObjectAddress functionAddress = GetObjectAddressFromParseTree((Node *) stmt, false);
 
-	List *commands = list_make4(DISABLE_DDL_PROPAGATION,
-								GetFunctionDDLCommand(address.objectId, true),
-								GetFunctionAlterOwnerCommand(address.objectId),
-								ENABLE_DDL_PROPAGATION);
+	if (IsObjectAddressOwnedByExtension(&functionAddress, NULL))
+	{
```

Comment:
On `create_distributed_function` we seem to call `EnsureExtensionFunctionCanBeDistributed` when `IsObjectAddressOwnedByExtension` is true.

Do we need that here as well?